HiP-EventStoreLib/ResourceType.cs

```diff
@@ -0,0 +1,53 @@
+﻿using MongoDB.Bson.Serialization.Attributes;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace PaderbornUniversity.SILab.Hip.EventSourcing
+{
+    public class ResourceType : IEquatable<ResourceType>
+    {
+        private static Dictionary<string, ResourceType> _dictionary = new Dictionary<string, ResourceType>();
+        public static IReadOnlyDictionary<string, ResourceType> ResourceTypeDictionary => _dictionary;
+
+        /// <summary>
+        /// This name is used in two ways:
+        /// 1) as a "type"/"kind of resource" identifier in events
+        /// 2) as the collection name in the MongoDB cache database
+        /// </summary>
+        [BsonElement]
+        public string Name { get; }
+
+        public Type Type { get; private set; }
```

Comment:
I know we talked about that already... Of course, `Type` can't be JSON- or BSON-serialized, but could we maybe assign the correct type on deserialization by looking it up in `_dictionary`? Otherwise people might be surprised that `Type` is null on a deserialized `ResourceType`-object.